spikes/lk_sensors_imu_lsm6dsox_fusion_calibration/main.cpp

```diff
@@ -0,0 +1,143 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cinttypes>
+
+#include "drivers/HighResClock.h"
+#include "rtos/ThisThread.h"
+
+#include "CoreI2C.h"
+#include "CoreLSM6DSOX.hpp"
+#include "LogKit.h"
+#include "fusion/Fusion.h"
+
+using namespace std::chrono;
+using namespace leka;
+
+namespace {
+
+namespace imu {
+
+	namespace internal {
+
+		auto drdy_irq = CoreInterruptIn {PinName::SENSOR_IMU_IRQ};
+		auto i2c	  = CoreI2C(PinName::SENSOR_IMU_TH_I2C_SDA, PinName::SENSOR_IMU_TH_I2C_SCL);
+
+	}	// namespace internal
+
+	CoreLSM6DSOX lsm6dsox(internal::i2c, internal::drdy_irq);
+
+}	// namespace imu
+
+namespace fusion {
+
+	constexpr auto kODR_HZ = int {52};
+
+	auto ahrs = FusionAhrs {};
+
+	const FusionAhrsSettings settings = {
+		.gain				   = 0.5F,
+		.accelerationRejection = 10.0F,
+		.magneticRejection	   = 0.0F,
+		.rejectionTimeout	   = static_cast<unsigned int>(5 * kODR_HZ),   // ? # of samples in 5 seconds
+	};
```

Comment:
This answer to my previous question. Seems those parameters to have an influence on results or this is insignificant compared to calibration for instance ?

Reply:
they don't work on the same things.

calibration is the first thing that you need. working with uncalibrated values is like working with no values at all.

Once you have calibrated values, you can improve the results better.